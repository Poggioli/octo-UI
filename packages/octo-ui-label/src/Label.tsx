import { CSS, forwardRef } from "@octo-ui/core";
import { StyledText } from "@octo-ui/text";
import { StyledLabel } from "./styles";
import { LabelProps } from "./types";

const Label = forwardRef<typeof StyledLabel, LabelProps>(
  ({ color, ...props }, forwardedRef) => {
    const css: CSS = {
      color: color ?? "$grey12",
      ...props.css,
    };

    return (
      <StyledText as={StyledLabel} {...props} ref={forwardedRef} css={css} />
    );
  }
);
Label.toString = () => `.${StyledLabel.className}`;

export { StyledLabel };
export { Label };
