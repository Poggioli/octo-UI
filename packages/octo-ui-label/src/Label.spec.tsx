import { render } from "@testing-library/react";
import { Label } from "./Label";

describe("<Label />", () => {
  it(`GIVEN a <Label />
      WHEN render is called
      THEN should render a <Label />`, () => {
    const rendered = render(<Label>Label</Label>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Label />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Label.toString()).toBe(".octo-UI-c-gHWBzz");
  });

  describe("color props", () => {
    it(`GIVEN a <Label />
        WHEN color props is not setted
        THEN should render a <Label /> $grey12 as color`, () => {
      const rendered = render(<Label>Label</Label>);
      expect(rendered.container).toMatchSnapshot();
    });

    it(`GIVEN a <Label />
        WHEN color props is $grey11
        THEN should render a <Label /> $grey11 as color`, () => {
      const rendered = render(<Label color="$grey11">Label</Label>);
      expect(rendered.container).toMatchSnapshot();
    });
  });
});
