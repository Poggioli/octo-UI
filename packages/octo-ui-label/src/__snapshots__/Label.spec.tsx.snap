// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> GIVEN a <Label />
      WHEN render is called
      THEN should render a <Label /> 1`] = `
<div>
  <label
    class="octo-UI-c-gHWBzz octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Label
  </label>
</div>
`;

exports[`<Label /> color props GIVEN a <Label />
        WHEN color props is $grey11
        THEN should render a <Label /> $grey11 as color 1`] = `
<div>
  <label
    class="octo-UI-c-gHWBzz octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-ibuyHBD-css"
  >
    Label
  </label>
</div>
`;

exports[`<Label /> color props GIVEN a <Label />
        WHEN color props is not setted
        THEN should render a <Label /> $grey12 as color 1`] = `
<div>
  <label
    class="octo-UI-c-gHWBzz octo-UI-c-gWerhV octo-UI-c-gWerhV-iMVRpC-size-3 octo-UI-c-gWerhV-iisyIHw-css"
  >
    Label
  </label>
</div>
`;
